test/dateTime.js

```diff
@@ -15,6 +17,286 @@ describe('dateTime', () => {
 
 	afterEach(() => documentLocaleSettings.reset());
 
+	describe('converting date/time', () => {
+
+		function getExpectedResult(date, minutes, hours, offset) {
+			// calculate what expected hours and minutes should be based on offset
+			const offsetHours = Math.floor(Math.abs(offset));
+			if (offset < 0) {
+				hours -= offsetHours;
+			} else {
+				hours += offsetHours;
+			}
+			if ((offset % 1) !== 0) {
+				minutes = minutes + 60 * (offset % 1);
+				if (minutes > 60) {
+					minutes -= 60;
+					hours += 1;
+				}
+				hours = Math.floor(hours);
+			}
+
+			// adjust date based on number of hours
+			if (hours < 0) {
+				hours = 24 + hours;
+				date--;
+			} else if (hours >= 24) {
+				hours -= 24;
+				date++;
+			}
+
+			return {
+				date: date,
+				hours: hours,
+				minutes: minutes
+			};
+		}
+
+		describe('convertUTCToLocalDateTime', () => {
+			it('should return original date if timezone identifier is blank', () => {
+				const UTCDateTime = {
+					month: 8,
+					date: 25,
+					year: 2015,
+					hours: 12,
+					minutes: 28,
+					seconds: 0
+				};
+
+				documentLocaleSettings.timezone.identifier = '';
+				const result = convertUTCToLocalDateTime(UTCDateTime);
+				expect(result).to.deep.equal(UTCDateTime);
+			});
+
+			it('should throw if timezone identifier is invalid', () => {
+				const UTCDateTime = {
+					month: 8,
+					date: 25,
+					year: 2015,
+					hours: 12,
+					minutes: 28,
+					seconds: 0
+				};
+
+				documentLocaleSettings.timezone.identifier = 'FAKE';
+				expect(() => {
+					convertUTCToLocalDateTime(UTCDateTime);
+				}).to.throw();
+			});
+
+			[
+				{timezone: 'Pacific/Rarotonga', expectedGMTOffset: -10},
+				{timezone: 'America/Santa_Isabel', expectedGMTOffset: -7},
+				{timezone: 'America/Toronto', expectedGMTOffset: -4},
+				{timezone: 'Atlantic/Reykjavik', expectedGMTOffset: 0},
+				{timezone: 'Australia/Eucla', expectedGMTOffset: 8.75},
+				{timezone: 'Australia/Darwin', expectedGMTOffset: 9.5},
+				{timezone: 'Pacific/Apia', expectedGMTOffset: 13}
+			].forEach((test) => {
+				it(`should have expected GMT offset of ${test.expectedGMTOffset} for timezone ${test.timezone}`, () => {
+					const hours = 12;
+					const minutes = 28;
+					const date = 25;
+					const UTCDateTime = {
+						month: 8,
+						date: date,
+						year: 2015,
+						hours: hours,
+						minutes: minutes,
+						seconds: 0
+					};
+
+					documentLocaleSettings.timezone.identifier = test.timezone;
+					const result = convertUTCToLocalDateTime(UTCDateTime);
+					const expectedResult = getExpectedResult(date, minutes, hours, test.expectedGMTOffset);
```

Comment:
Just thinking... if `getExpectedResult()` took in the whole input value and returned a full output value, then the tests could be simplified to:

```
const input = ...;
documentLocaleSettings.timezone.identifier = test.timezone;
const result = convertUTCToLocalDateTime(input);
const expected = getExpectedResult(input, test.expectedGMTOffset);
expect(result).to.deep.equal(expected);
```